.github/resources/check_flakes.py

```diff
@@ -0,0 +1,286 @@
+        .ewm(alpha=EWM_ALPHA, adjust=EWM_ADJUST)
+        .mean()
+        .droplevel("test_identifier")
+    )
+
+    return fliprate_table[fliprate_table.flip_rate != 0]
+
+def calculate_n_runs_fliprate_table(testrun_table: pd.DataFrame, window_size: int) -> pd.DataFrame:
+    """Calculate fliprates for given n run window and select m of those windows
+    Return a table containing the results.
+    """
+    # Apply non_overlapping_window_fliprate to each group in testrun_table
+    fliprates = testrun_table.groupby("test_identifier")["test_status"].apply(
+        lambda x: non_overlapping_window_fliprate(x, window_size)
+    )
+
+    # Convert fliprates Series of DataFrames to a DataFrame
+    fliprate_table = fliprates.reset_index()
+
+    # Rename the columns in fliprate_table
+    fliprate_table = fliprate_table.rename(columns={"flip_rate": "flip_rate", "consecutive_failures": "consecutive_failures"})
+    print("Fliprate table of all the tests:")
+    print(fliprate_table)
+
+    # Calculate the EWMA of flip rates for each test identifier
+    fliprate_table["flip_rate_ewm"] = (
+        fliprate_table.groupby("test_identifier")["flip_rate"]
+        .ewm(alpha=EWM_ALPHA, adjust=EWM_ADJUST)
+        .mean()
+        .droplevel("test_identifier")
+    )
+
+    # Rename the index level to "window"
+    fliprate_table = fliprate_table.rename(columns={"level_1": "window"})
+
+    # Filter out rows where flip_rate is not zero
+    fliprate_table = fliprate_table[fliprate_table.flip_rate != 0]
+
+    return fliprate_table
+
+def get_top_fliprates(fliprate_table: pd.DataFrame, top_n: int, precision: int) -> Dict[str, tuple[Decimal, int]]:
+    """Return the top n highest scoring test identifiers and their scores
+
+    Look at the last calculation window for each test from the fliprate table
+    and return the top n highest scoring test identifiers along with their scores
+    and corresponding consecutive failures.
+    """
+    context = getcontext()
+    context.prec = precision
+    context.rounding = ROUND_UP
+    last_window_values = fliprate_table.groupby("test_identifier").last()
+
+    if top_n != 0:
+        top_fliprates_ewm = last_window_values.nlargest(top_n, "flip_rate_ewm")
+    else :
+        top_fliprates_ewm = last_window_values.nlargest(len(last_window_values), "flip_rate_ewm")
+
+    # Create a dictionary with test_identifier as keys and a tuple of flip_rate_ewm and consecutive_failures as values
+    results = {}
+    for test_id, row in top_fliprates_ewm.iterrows():
+        results[test_id] = (Decimal(row["flip_rate_ewm"]), row["consecutive_failures"])
+
+    return results
+
+
+def parse_junit_suite_to_df(suite: TestSuite) -> list:
+    """Parses Junit TestSuite results to a test history dataframe"""
+    dataframe_entries = []
+    time = suite.attrib.get('timestamp')
+
+    for testcase in suite.findall('.//testcase'):
+        test_identifier = testcase.attrib.get('classname') + "::" + testcase.attrib.get('name')
```

Comment:
The class name is just noise IMO
```suggestion
        test_identifier = testcase.attrib.get('name')
```